jaxrs2/src/main/java/feign/jaxrs2/JAXRS2Contract.java

```diff
@@ -13,25 +13,99 @@
  */
 package feign.jaxrs2;
 
-import javax.ws.rs.BeanParam;
+import feign.jaxrs.JAXRSContract;
+import javax.ws.rs.*;
 import javax.ws.rs.container.Suspended;
 import javax.ws.rs.core.Context;
-import feign.jaxrs.JAXRSContract;
+import java.lang.reflect.Field;
+import static feign.Util.checkState;
+import static feign.Util.emptyToNull;
 
 /**
  * Please refer to the <a href="https://github.com/Netflix/feign/tree/master/feign-jaxrs2">Feign
  * JAX-RS 2 README</a>.
  */
 public final class JAXRS2Contract extends JAXRSContract {
 
+  private final boolean enableBeanParamSupport;
+
   public JAXRS2Contract() {
+    this(false);
+  }
+
+  public JAXRS2Contract(boolean enableBeanParamSupport) {
+    this.enableBeanParamSupport = enableBeanParamSupport;
     // parameter with unsupported jax-rs annotations should not be passed as body params.
     // this will prevent interfaces from becoming unusable entirely due to single (unsupported)
     // endpoints.
     // https://github.com/OpenFeign/feign/issues/669
     super.registerParameterAnnotation(Suspended.class, (ann, data, i) -> data.ignoreParamater(i));
     super.registerParameterAnnotation(Context.class, (ann, data, i) -> data.ignoreParamater(i));
-    super.registerParameterAnnotation(BeanParam.class, (ann, data, i) -> data.ignoreParamater(i));
```

Comment:
I think, before made sense to ignore any `@BeanParam` cause it was not implemented.

Now that you fixed that shortcoming, I think it's fine to have it enabled

Reply:
Let me know if you would like me to make this change.

Reply:
I think is a nice to have, but your call

Reply:
Added and pushed this change.

Reply:
Pushed this change. Jdk8 appears to be failing again in CI. Not able to duplicate on my local